strawberryfields/io.py

```diff
@@ -83,7 +84,11 @@ def to_blackbird(prog, version="1.0"):
             for a in cmd.op.p:
                 if sfpar.par_is_symbolic(a):
                     # SymPy object, convert to string
-                    a = str(a)
+                    if re.search(r"(?<!\w)q\d*(?!\w)", str(a)):
```

Comment:
I was wondering if we could do this without having to run a regex on its string form.  Maybe something like:

```
isMeasuredParameter = lambda x : isinstance(x, sf.parameter.MeasuredParameter)
if any(map(isMeasuredParameter, x.free_symbols)):
```

Maybe I just have personal biases against regular expressions.

Reply:
Nice, yes, this is a much nicer solution. Thanks @albi3ro for catching this! 🥇 💯